suites/apis/clientCredGrantTest.js

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable max-len */
+
+/*
+
+1. create an OIDC client with the client_credential grant
+fence-create with --grant-types `client_credentials`
+which return client_id and secret_id
+
+2. run usersync (The client will have been previously added to the Jenkins user.yaml and granted access to
+what it will need for the test, but i think usersync will not grant the access if the client does not
+exist in the Fence DB. So you need to run usersync again during the test, after creating the client)
+
+3. get a client access token using the client ID and secret.
+
+4. create an access request in Requestor
+
+5. list Requestor requests
+
+6. update an access request in Requestor
+
+7. AfterSuite - delete the create client in step 1 with `client-delete` command
+also delete the requests created in requestor
+*/
+
+Feature('Client_Credentials Grant Type @requires-fence');
+
+const { expect } = require('chai');
+const { runUserSync, checkPod, getAccessTokenHeader } = require('../../utils/apiUtil.js');
+const { Bash } = require('../../utils/bash.js');
+const requestorTasks = require('../../services/apis/requestor/requestorTasks.js');
+const requestorProps = require('../../services/apis/requestor/requestorProps.js');
+
+const bash = new Bash();
+
+const clientName = 'jenkinsClientTester';
+
+async function getAccessToken(clientID, secretID) {
+  const tokenReq = bash.runCommand(`curl --user "${clientID}:${secretID}" --request POST "https://${process.env.HOSTNAME}/user/oauth2/token?grant_type=client_credentials" -d scope="openid user"`);
+  if (process.env.DEBUG === 'true') {
+    console.log(`data : ${tokenReq}`);
+  }
+  const tokens = JSON.parse(tokenReq);
+  const accessToken = tokens.access_token;
+  if (process.env.DEBUG === 'true') {
+    console.log(`###Access_Token : ${accessToken}`);
+  }
+  return accessToken;
+}
+
+BeforeSuite(async ({ I, fence }) => {
+  I.cache = {};
+  fence.do.deleteClient(clientName);
+});
+
+AfterSuite(async ({ I, fence }) => {
+  // deleting the request created
+  console.log('Deleting the request from the requestor DB ... ');
+  if (process.env.DEBUG === 'true') {
+    console.log(I.cache.requestID);
+  }
+  const deleteRequest = await requestorTasks.deleteRequest(I.cache.requestID);
+  if (deleteRequest.status === 200) {
+    console.log(`Request ${I.cache.requestID} deleted successfully`);
+  }
+  // delete the client jenkinsClientTester
+  console.log('Deleting client created for the test ...');
+  fence.do.deleteClient(clientName);
+
+  // revoking the arborist policy for the user
+  console.log('Revoking the arborist policy for the user0 ...');
+  await bash.runCommand(`
+      gen3 devterm curl -X DELETE arborist-service/user/dcf-integration-test-0@planx-pla.net/policy/requestor_integration_test
```

Comment:
can we move this to a separate service (arborist) since this is basically an arborist api call?

Reply:
Currently this is the only way  of doing it. Not all the endpoints are exposed via revproxy as [delete endpoint](https://petstore.swagger.io/?url=https://raw.githubusercontent.com/uc-cdis/arborist/master/docs/openapi.yaml#/user/delete_user__username__policy__policyName_) does not have an auth layer. Also the revproxy [conf](https://github.com/uc-cdis/cloud-automation/blob/master/kube/services/revproxy/gen3.nginx.conf/arborist-service.conf) for arborist